scripts/cleanup_edomain_templates.py

```diff
@@ -0,0 +1,252 @@
+    return args
+
+
+def make_ssh_client(rhevip, sshname, sshpass):
+    connect_kwargs = {
+        'username': sshname,
+        'password': sshpass,
+        'hostname': rhevip
+    }
+    return SSHClient(**connect_kwargs)
+
+
+# get the domain edomain path on the rhevm
+def get_edomain_path(api, edomain):
+    edomain_id = api.storagedomains.get(edomain).get_id()
+    edomain_conn = api.storagedomains.get(edomain).storageconnections.list()[0]
+    return (edomain_conn.get_path() + '/' + edomain_id,
+            edomain_conn.get_address())
+
+
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
+    """Cleanup all the empty directories on the edomain/edomain_id/master/vms
+    else api calls will result in 400 Error with ovf not found,
+
+    Args:
+        path: path for vms directory on edomain.
+        edomain: Export domain of chosen RHEVM provider.
+        edomainip: edomainip to connect through ssh.
+        sshname: edomain ssh credentials.
+        sshpass: edomain ssh credentials.
+    """
+    print "RHEVM: Deleting the empty directories on edomain/vms file..."
+    ssh_client = make_ssh_client(edomainip, sshname, sshpass)
+    command = 'cd {}/master/vms && find . -maxdepth 1 -type d -empty -delete'.format(path)
+    exit_status, output = ssh_client.run_command(command)
+    if exit_status != 0:
+        print "RHEVM: Error while deleting the empty directories on path.."
+        print output
+
+
+def is_edomain_template_deleted(api, name, edomain):
+    """Checks for the templates delete status on edomain.
+
+    Args:
+        api: API for RHEVM.
+        name: template_name
+        edomain: Export domain of selected RHEVM provider.
+    """
+    return not api.storagedomains.get(edomain).templates.get(name)
+
+
+def delete_edomain_templates(api, template, edomain):
+    """deletes the template on edomain.
+
+    Args:
+        api: API for RHEVM.
+        name: template_name
+        edomain: Export domain of selected RHEVM provider.
+    """
+    with lock:
+        creation_time = template.get_creation_time().strftime("%d %B-%Y")
+        name = template.get_name()
+        print 'Deleting {} created on {} ...'.format(name, creation_time)
+    try:
+        template.delete()
+        print 'waiting for {} to be deleted..'.format(name)
+        wait_for(is_edomain_template_deleted,
+                 [api, name, edomain], fail_condition=False, delay=5)
+        print "RHEVM: successfully deleted the template {}".format(name)
+    except Exception as e:
+        with lock:
+            print "RHEVM: Exception occurred while deleting the template {}".format(name)
+            logger.exception(e)
+
+
+def cleanup_templates(api, edomain, days, max_templates):
+    templates = api.storagedomains.get(edomain).templates.list()
+    thread_queue = []
+    delete_templates = []
+    for template in templates:
+        delta = datetime.timedelta(days=days)
+        now = datetime.datetime.now().isoformat()
```

Comment:
AFAIK `.isoformat()` is yet another way how to format the datetime into strings. I believe that it should work even without formatting it as strings since the datetime data structures are designed to be compared and should know it even better than strings :). The `.get_creation_time()` return timezone-aware datetime. Local, timezone aware datetime can be obtained like this: http://stackoverflow.com/a/4530166 . So, these two should cooperate together well.